File: solutions/java/hamming/1/src/main/java/Hamming.java
class Hamming {
    String leftStrand;
    String rightStrand;

    Hamming(String leftStrand, String rightStrand) {
        if (leftStrand.length() == 0 && rightStrand.length() > 0)
            throw new IllegalArgumentException("left strand must not be empty.");
        if (leftStrand.length() > 0 && rightStrand.length() == 0)
            throw new IllegalArgumentException("right strand must not be empty.");
        if (leftStrand.length() != rightStrand.length())
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
        this.leftStrand = leftStrand;
        this.rightStrand = rightStrand;
    }

    int getHammingDistance() {
        int totalDifferences = 0;
        for (int i = 0; i < this.leftStrand.length(); i++) {
            if (this.leftStrand.charAt(i) != this.rightStrand.charAt(i))
                totalDifferences++;
        }
        return totalDifferences;
    }

}

File: solutions/typescript/hamming/1/hamming.ts
export function compute(left: string, right: string): number {
  if (left.length !== right.length) throw new Error("DNA strands must be of equal length.");

  let difference = 0;
  for (let i = 0; i < left.length; i++){
    if (left.charAt(i) != right.charAt(i)) difference++;
  }
  
  return difference;
}
